engine/src/main/java/org/terasology/rendering/primitives/BlockMeshGeneratorSingleShape.java

```diff
@@ -142,6 +142,10 @@ private boolean isSideVisibleForBlockTypes(Block blockToCheck, Block currentBloc
             return false;
         }
 
+        if (currentBlock.isWater() && (side == Side.TOP) && !blockToCheck.isWater()){
```

Comment:
Please add a `//TODO: ...` comment here to indicate that this only fixes part of a broader problem:

```suggestion
        //TODO: This only fixes the "water under block" issue of the top side not being rendered.
        //      Find a more general way to handle this for "engine:loweredCube" shapes.
        if (currentBlock.isWater() && (side == Side.TOP) && !blockToCheck.isWater()){
```

Reply:
Thanks Skaldarnar, good point; I've added the TODO. I also put a note about IntelliJ warning `!blockToCheck.isLiquid()` always returns true